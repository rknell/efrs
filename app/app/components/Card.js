import React, { Component } from 'react'
import { Text, TouchableOpacity, View } from 'react-native'
import { observer } from 'mobx-react/native'
import Fields from './Fields'

@observer
class Card extends Component {
  add () {
    this.props.values.push(JSON.parse(JSON.stringify(this.props.fields)))
  }

  render () {
    return (
      <View style={{ backgroundColor: 'rgba(0,0,0,0.1)' }}>
        <View style={{ flexDirection: 'row', alignItems: 'center', paddingLeft: 10, paddingRight: 10 }}>
          <Text style={{ flex: 1 }}>{this.props.label}</Text>
          <TouchableOpacity onPress={this.add.bind(this)}><Text style={{ fontSize: 30 }}>+</Text></TouchableOpacity>
        </View>

        {this.props.values && this.props.values.map((item, index) => {
          return (
            <View key={index} style={{
              borderWidth: 1,
              padding: 3,
              margin: 5,
              elevation: 2,
              borderColor: 'rgba(0,0,0,0.1)',
              backgroundColor: 'white'
            }}><Fields fields={item} /></View>
          )
        })}
        {(!this.props.values || !!this.props.values.length) &&
        <View style={{ flexDirection: 'row', alignItems: 'center', paddingLeft: 10, paddingRight: 10 }}>
          <Text style={{ flex: 1 }}>Add Another</Text>
          <TouchableOpacity onPress={this.add.bind(this)}><Text style={{ fontSize: 30 }}>+</Text></TouchableOpacity>
        </View>
        }
      </View>
    )
  }
}

export default Card
