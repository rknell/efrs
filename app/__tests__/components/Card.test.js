import React from 'react'
import Card from '../../app/components/Card'
import { shallow } from 'enzyme'
import toJSON from 'enzyme-to-json'

it('matches snapshot', () => {
  const wrapper = shallow(
    <Card />
  )
  expect(toJSON(wrapper)).toMatchSnapshot()
  wrapper.setProps({ count: 1 })
  expect(toJSON(wrapper)).toMatchSnapshot()
})

it('handles adding', () => {
  const wrapper = shallow(<Card values={[
    { test: true }
  ]} fields={{ test: true }} />)

  const instance = wrapper.instance()
  instance.add()
})
