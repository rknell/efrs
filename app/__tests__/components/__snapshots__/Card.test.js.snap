// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<View
  style={
    Object {
      "backgroundColor": "rgba(0,0,0,0.1)",
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "paddingLeft": 10,
        "paddingRight": 10,
      }
    }
  >
    <Text
      style={
        Object {
          "flex": 1,
        }
      }
    />
    <TouchableOpacity
      activeOpacity={0.2}
      onPress={[Function]}
    >
      <Text
        style={
          Object {
            "fontSize": 30,
          }
        }
      >
        +
      </Text>
    </TouchableOpacity>
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "paddingLeft": 10,
        "paddingRight": 10,
      }
    }
  >
    <Text
      style={
        Object {
          "flex": 1,
        }
      }
    >
      Add Another
    </Text>
    <TouchableOpacity
      activeOpacity={0.2}
      onPress={[Function]}
    >
      <Text
        style={
          Object {
            "fontSize": 30,
          }
        }
      >
        +
      </Text>
    </TouchableOpacity>
  </View>
</View>
`;

exports[`matches snapshot 2`] = `
<View
  style={
    Object {
      "backgroundColor": "rgba(0,0,0,0.1)",
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "paddingLeft": 10,
        "paddingRight": 10,
      }
    }
  >
    <Text
      style={
        Object {
          "flex": 1,
        }
      }
    />
    <TouchableOpacity
      activeOpacity={0.2}
      onPress={[Function]}
    >
      <Text
        style={
          Object {
            "fontSize": 30,
          }
        }
      >
        +
      </Text>
    </TouchableOpacity>
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "paddingLeft": 10,
        "paddingRight": 10,
      }
    }
  >
    <Text
      style={
        Object {
          "flex": 1,
        }
      }
    >
      Add Another
    </Text>
    <TouchableOpacity
      activeOpacity={0.2}
      onPress={[Function]}
    >
      <Text
        style={
          Object {
            "fontSize": 30,
          }
        }
      >
        +
      </Text>
    </TouchableOpacity>
  </View>
</View>
`;
